src/frontend/js/components/PaymentButton/index.spec.tsx

```diff
@@ -0,0 +1,451 @@
+  });
+
+  it('should render a payment button with a specific label when a credit card is provided', () => {
+    /*
+      If a credit card is provided, it seems that the payment should be a one click,
+      so the payment button label should mention this information.
+    */
+    const product = ProductFactory.generate();
+    const creditCard = CreditCardFactory.generate();
+
+    render(
+      <Wrapper>
+        <PaymentButton product={product} creditCard={creditCard} onSuccess={jest.fn()} />
+      </Wrapper>,
+    );
+
+    const $button = screen.getByRole('button', {
+      name: `Pay in one click ${formatPrice(product.price, product.price_currency)}`,
+    }) as HTMLButtonElement;
+
+    // As a billing address is missing, button should be disabled
+    expect($button.disabled).toBe(true);
+  });
+
+  it('should render an enabled payment button if a billing address is provided', () => {
+    const product = ProductFactory.generate();
+    const billingAddress = AddressFactory.generate();
+
+    render(
+      <Wrapper>
+        <PaymentButton billingAddress={billingAddress} product={product} onSuccess={jest.fn()} />
+      </Wrapper>,
+    );
+
+    const $button = screen.getByRole('button', {
+      name: `Pay ${formatPrice(product.price, product.price_currency)}`,
+    }) as HTMLButtonElement;
+
+    // As a billing address is missing, button should be disabled
+    expect($button.disabled).toBe(false);
+  });
+
+  it('should create an order then display the payment interface when user clicks on payment button', async () => {
+    const product = ProductFactory.generate();
+    const billingAddress = AddressFactory.generate();
+    const order = OrderWithPaymentFactory.generate();
+    const handleSuccess = jest.fn();
+
+    fetchMock
+      .post('https://joanie.test/api/orders/', order)
+      .get(`https://joanie.test/api/orders/${order.id}/`, {
+        ...order,
+        state: OrderState.PENDING,
+      });
+
+    const { clientState } = PersistedClientFactory({
+      queries: [QueryStateFactory('user', { data: { username: 'John Doe' } })],
+    });
+    const client = createQueryClient();
+    hydrate(client, clientState);
+
+    render(
+      <Wrapper client={client}>
+        <PaymentButton
+          billingAddress={billingAddress}
+          product={product}
+          onSuccess={handleSuccess}
+        />
+      </Wrapper>,
+    );
+
+    const $button = screen.getByRole('button', {
+      name: `Pay ${formatPrice(product.price, product.price_currency)}`,
+    }) as HTMLButtonElement;
+
+    // - As all information are provided, payment button should not be disabled.
+    expect($button.disabled).toBe(false);
+
+    // - User clicks on pay button
+    await act(async () => {
+      fireEvent.click($button);
+    });
+
+    // - Route to create order should have been called
+    expect(fetchMock.calls()).toHaveLength(1);
+    expect(fetchMock.lastUrl()).toBe('https://joanie.test/api/orders/');
```

Comment:
In general, shouldn't we check it has been called with expected params and use toHaveBeenCalledWith

Reply:
This is true for mock using `jest.fn()` but in the case of `fetchMock` it gives access to a method to retrieve all calls executed and some methods to retrieve informations about the last call.

Reply:
I mean, to do a full test, I would test the post sent to the API are correct, I don't see this kind of check-in the test.

Reply:
Oh ok ! Yes I should be able to check the body structure of post requests.

Reply:
@carofun I add assertion to check body content of each post request.

Reply:
I like it!